src/sentry/integrations/webhook.py

```diff
@@ -0,0 +1,52 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any
+
+from django.views.decorators.csrf import csrf_exempt
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import Endpoint
+
+logger = logging.getLogger(__name__)
+
+
+class IntegrationWebhookEndpoint(Endpoint, ABC):
+    owner = ApiOwner.ECOSYSTEM
+    publish_status = {
+        "POST": ApiPublishStatus.PRIVATE,
+    }
+    authentication_classes = ()
+    permission_classes = ()
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+        self.log_extra = {}
```

Comment:
While this is helpful for consistent logging, do we always need to log this full context, or is it ok to have the subclass methods decide what should be logged?

Reply:
the only logs that are by default is `request_method` and `request_path`, and each subclass adds more into `self.log_extra` as necessary. what do you mean by full context here?

Reply:
Just that we're always appending additional data into this context object. If these are the only 2 places we're logging though, it's fine.

Reply:
oh wait i think i have to double check that this is actually being initialized each time. maybe i reinitialize when dispatch is called?